### PR TITLE
WC-169: swallow input if value takes in an empty string

### PR DESCRIPTION
**Pull request by browne0**
#### Related issues
Fixes https://meetup.atlassian.net/browse/WC-169

#### Description
In a time input component, the time is first set as an empty string. We only care about the valid time, so return out.

Reference: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/time#Validation

Links back to https://github.com/meetup/mup-web/pull/2540

#### Screenshots (if applicable)
bug

### CODE REVIEW

File: src/forms/TimeInput.jsx [hunk 1]
```diff
@@ -20,56 +20,56 @@ const formatDigits = (number) => `0${number}`.slice(-2);
  * @returns {String|Object} the only parsed value or all the parsed parts if no part defined
  */
 export const getTimeParts = (time) => {
-	const [ hours, minutes ] = time.split(':');
-	return {
-		hours,
-		minutes,
-		meridian: hours < 12 ? 'AM': 'PM',
-	};
+  const [hours, minutes] = time.split(':');
```
Comment (mmcgahan) [inline src/forms/TimeInput.jsx]: looks like some bad Prettier reformatting - everything should remain tabs, not spaces - make sure you check your PRs when you first set them up to catch little things like this

File: src/forms/TimeInput.jsx [hunk 2]
```diff
@@ -78,321 +78,326 @@ export class TimeInput extends React.Component {
...
 	* @param e Event Object
 	* @description called when the browser-native time input changes, in turn calls the onChange
 	* 	handler prop, if there is one provided (eg supplied by redux-form or DateTimePicker)
 	*/
-	onTimeInputChange(e) {
-		const { value } = e.target;
-		this.setState(() => ({ value }));
+  onTimeInputChange(e) {
+    const { value } = e.target;
+    // In a time input component, the time is first set as an empty string. We only care about the valid time, so return out.
```
Comment (mmcgahan) [inline src/forms/TimeInput.jsx]: What does 'the time is first set as an empty string' mean? In the problem we were seeing, the `value` attribute of the element was already set (e.g. `<input type="time" value="18:00" />`, so it definitely wasn't empty at first. The problem we were seeing was `value` getting set to the empty string _when a user manually entered `0` in the hours field_ - something like `'time inputs are allowed to return an empty string for invalid input - ignore all invalid input'`

sort of nitpick but 'return out' is sort of an ambiguous turn of phrase. 'return immediately' or 'exit early' are a bit more clear.

File: src/forms/TimeInput.jsx (hunk 1 above)
Comment (browne0) [reply src/forms/TimeInput.jsx]: Fixed it, that was an editor setting. I assumed it had the pre commit hooks to reformat like mup-web, my bad.

File: src/forms/TimeInput.jsx (hunk 2 above)
Comment (browne0) [reply src/forms/TimeInput.jsx]: done

File: src/forms/TimeInput.jsx [hunk 3]
```diff
@@ -20,56 +20,56 @@ const formatDigits = (number) => `0${number}`.slice(-2);
  * @returns {String|Object} the only parsed value or all the parsed parts if no part defined
  */
 export const getTimeParts = (time) => {
-	const [ hours, minutes ] = time.split(':');
-	return {
-		hours,
-		minutes,
-		meridian: hours < 12 ? 'AM': 'PM',
-	};
+    const [hours, minutes] = time.split(':');
```
Comment (mmcgahan) [inline src/forms/TimeInput.jsx]: still using spaces, not tabs

File: src/forms/TimeInput.jsx [hunk 4]
```diff
@@ -100,6 +100,11 @@ export class TimeInput extends React.Component {
 	*/
 	onTimeInputChange(e) {
 		const { value } = e.target;
+		// Time inputs are allowed to return an empty string for invalid input.
+		// We should ignore all invalid input and return immediately
+		// Reference: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/time#Validation
+		if (value === '') return;
```
Comment (aquillano) [inline src/forms/TimeInput.jsx]: 👍

**Pull request merged**